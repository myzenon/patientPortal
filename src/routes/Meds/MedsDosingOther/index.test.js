import React from 'react'
import { MedsDosingOther } from './index'

import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: { "frequency": "asNeeded" }
            }
        },
        
        match: {
            params: {
                medId: '000080833'
            }
        },
        page: {
            data: {}
        }
    }
    const wrapper = render(
        <MedsDosingOther { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})
test.skip('medication image comes from cmr.medications[i].image.small.url', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: { "frequency": "asNeeded" }
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        page: {
            data: {}
        }
    }
    const wrapper = mount(
        <MedsDosingOther { ...props } />
    )
    setTimeout(() => {
        wrapper.update()
        expect(wrapper.find('img.contentImg').props().src).toBe(mockObj.cmr.medications[0].image.small.url)
        done()
    }, 0)
})

test('medication name comes from cmr.medications[i].display', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: { "frequency": "asNeeded" }
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        page: {
            data: {}
        }
    }
    const wrapper = mount(
        <MedsDosingOther { ...props } />
    )
    setTimeout(() => {
        expect(wrapper.find('.heading1').text()).toBe(mockObj.cmr.medications[0].display)
        done()
    }, 0)
})


test.skip('clicks next button without any input | api fhirCmrIdPut called | go to MedsWhatFor page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: { "frequency": "asNeeded" }
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        page: {
            data: {}
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({"medication_000080833": {"dosage": {"description": " ", "frequency": "asNeeded"}, "taken": "y"}})
            return Promise.resolve()
        },
        savePageData: (page) => {
            expect(page).toEqual({
            })
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "y",
                    dosage: {"description": " "}
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-what-for/000080833')
            done()
        }
    }
    const wrapper = mount(
        <MedsDosingOther  {...props} />
    )
    setTimeout(() => {
        wrapper.find('button[type="submit"]').simulate('submit')
    }, 0)
})

test.skip('clicks next button with inputting XX pills every XX hours | api fhirCmrIdPut called | go to MedsWhatFor page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: { "frequency": "asNeeded" }
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        page: {
            data: {}
        },
        savePageData: (page) => {
            expect(page).toEqual({
            })
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "y",
                    dosage: {
                            frequency: "2",
                            period: "day",
                            quantity: "6",
                            description: " "
                    }
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "y",
                    dosage: {
                            frequency: "2",
                            period: "day",
                            quantity: "6",
                            description: " "
                    }
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-what-for/000080833')
            done()
        }
    }
    const wrapper = mount(
        <MedsDosingOther  {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="frequency"]').simulate('change', {
            target: {
                name: 'frequency',
                value: '2'
            }
        })
        wrapper.find('input[name="quantity"]').simulate('change', {
            target: {
                name: 'quantity',
                value: '6'
            }
        })
        wrapper.find('button[type="submit"]').simulate('submit')
    }, 0)
})

test.skip('clicks next button with inputting XX pills a day | api fhirCmrIdPut called | go to MedsWhatFor page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: { "frequency": "asNeeded" }
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        page: {
            data: {
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "y",
                    dosage: {
                            frequency: "4",
                            period: "day",
                            quantity: "2",
                            description: " "
                        },
                    }
            })
            return Promise.resolve()
        },
        savePageData: (page) => {
            expect(page).toEqual({
            })
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "y",
                    dosage: {
                            frequency: "4",
                            period: "day",
                            quantity: "2",
                            description: " "
                        }
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-what-for/000080833')
            done()
        }
    }
    const wrapper = mount(
        <MedsDosingOther {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[name="quantity"]').simulate('change', {
            target: {
                name: 'quantity',
                value: '2'
            }
            })
            wrapper.find('input[name="frequency"]').simulate('change', {
                target: {
                    name: 'frequency',
                    value: '4'
                }
        })
        wrapper.find('button[type="submit"]').simulate('submit')
    }, 0)
})
