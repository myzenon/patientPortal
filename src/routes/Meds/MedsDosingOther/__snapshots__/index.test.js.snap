// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <span
      class="heading1"
    >
      EFFEXOR XR 75MG
    </span>
    <p />
    <p />
    <span
      class="questions"
    >
      Please tell us how you are taking this medication.
    </span>
    <p />
     
    <br />
    <form
      class="formWrapper"
    >
      <div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="inputDescTextWrapper"
            >
              <input
                autofocus=""
                class="inputTextCenter"
                name="description"
                placeholder="I take 1-2 tablets every 4-6 hours as needed for pain"
                textalign="center"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <br />
      <br />
      <button
        class="button btn btn-secondary"
        type="submit"
      >
        <b>
          NEXT
        </b>
      </button>
    </form>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepActive"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
